web-server/v0.4/mock/api.js

```diff
@@ -1,64 +1,64 @@
 import config from './datastoreConfig';
 
-export const mockControllerAggregation = {
-  aggregations: {
-    controllers: {
-      buckets: [
-        {
-          key: 'a_test_controller',
-          doc_count: 1,
-          runs_preV1: { value: null },
-          runs: { value: 1, value_as_string: '1111-11-11' },
-        },
-        {
-          key: 'b_test_controller',
-          doc_count: 2,
-          runs_preV1: { value: null },
-          runs: { value: 2, value_as_string: '2222-22-22' },
-        },
-      ],
+/* We generate 26 records to generate specific records starting
+ with each alphabet to assert our tests. Hence we have 26 instances
+ of indices and results as per the controllers. */
+
+// Generates mock controllers
+export const generateMockControllerAggregation = () => {
+  const mockControllerSchema = {
+    aggregations: {
+      controllers: {
+        buckets: [],
+      },
     },
-  },
+  };
+  for (let i = 0; i < 26; i += 1) {
+    mockControllerSchema.aggregations.controllers.buckets.push({
+      key: `${(i + 10).toString(36)}_test_controller`,
+      doc_count: i,
+      runs_preV1: { value: null },
+      runs: {
+        value: `${i + 1}`,
+        value_as_string: `${i + 1}${i + 1}${i + 1}${i + 1}-${i + 1}${i + 1}-${i + 1}${i + 1}`,
+      },
+    });
+  }
+  return mockControllerSchema;
 };
 
 const datastoreConfig = config['/dev/datastoreConfig'];
 const prefix = datastoreConfig.prefix + datastoreConfig.run_index.slice(0, -1);
-export const mockIndices = [
-  {
-    index: `${prefix}.0000-00-00`,
-  },
-  {
-    index: `${prefix}.0000-00-01`,
-  },
-];
+// Generates mock indices
+export const generateMockIndices = () => {
+  const indexSchema = [];
+  for (let i = 0; i < 26; i += 1) {
```

Comment:
Nice, this makes it clear that we are generating 26 items.

Now we need to not hard-code 26 and derive it from the total number of items we need for the pagination schemes we support.